torch/onnx/_internal/exporter.py

```diff
@@ -772,6 +781,54 @@ def save(
             else:
                 serializer.serialize(self, destination)
 
+    @_beartype.beartype
+    def save_sarif_log(self, destination: str) -> None:
+        """Saves the SARIF log to ``destination``.
+
+        Args:
+            destination: The destination to save the SARIF log. It must have a `.sarif` extension.
+        """
+        if not destination.endswith(".sarif"):
+            message = (
+                f"'destination' must have a .sarif extension, " f"got {destination}"
+            )
+            log.fatal(message)
+            raise ValueError(message)
+
+        self.diagnostic_context.dump(destination)
+
+    @classmethod
+    def from_export_failure(
```

Comment:
```suggestion
    def from_failure(
```
This is a method of `ExportOutput`, which implies `export` already.